crates/matrix-sdk-libolm-migration/src/types.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+/// Struct collecting data that is important to migrate sessions to the rust-sdk
+#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
```

Comment:
This is a simple move of these 5 types, with the only difference being that the `uniffi:*` derivation needs to become conditional on the feature flag.